common/src/main/java/org/apache/uniffle/common/ShuffleDataResult.java

```diff
@@ -51,16 +58,31 @@ public byte[] getData() {
     if (data.hasArray()) {
       return data.array();
     }
-    ByteBuffer dataBuffer = data.duplicate();
-    byte[] byteArray = new byte[dataBuffer.remaining()];
-    dataBuffer.get(byteArray);
-    return byteArray;
+    int offset = 0;
+    int length = calLength();
+    byte[] bytes = new byte[length];
+    if (data.nioBufferCount() == 1) {
+      ByteBuffer buffer = data.nioBuffer(data.readerIndex(), length);
+      buffer.get(bytes);
+    } else {
+      ByteBuffer[] buffers = data.nioBuffers(data.readerIndex(), length);
+      for (ByteBuffer buffer : buffers) {
+        int remaining = buffer.remaining();
+        buffer.get(bytes, offset, remaining);
+        offset += remaining;
+      }
+    }
+    return bytes;
   }
 
-  public ByteBuffer getDataBuffer() {
+  public ByteBuf getDataBuf() {
     return data;
   }
 
+  public ByteBuffer getDataBuffer() {
```

Comment:
Do you need to consider `nioBufferCount != 1`?

Reply:
It was my overlooked, since the increaseWriterIndex parameter is true when we call CompositeByteBuf.addComponent, CompositeByteBuf can be used as a normal ByteBuf.

Reply:
Do you mean that we don't need consider `nioBufferCount != 1`?

Reply:
Yes, we don't need, I have made sone tests.

Reply:
Yes, we don't need, I have made some tests.

Reply:
> Yes, we don't need, I have made some tests.

Could you add some UT for this case?

Reply:
See ByteBufUtilsTest#test, CompositeByteBuf can be read as ByteBuf.
```
final int expectedLength = byteBuf.readableBytes() + byteBuf1.readableBytes();
CompositeByteBuf compositeByteBuf = Unpooled.compositeBuffer();
compositeByteBuf.addComponent(true, byteBuf);
compositeByteBuf.addComponent(true, byteBuf1);

ByteBuf res = Unpooled.buffer(100);
ByteBufUtils.copyByteBuf(compositeByteBuf, res);
assertEquals(expectedLength, res.readableBytes() - Integer.BYTES);

res.clear();
ByteBufUtils.copyByteBuf(compositeByteBuf, res);
assertEquals(expectedLength, res.readableBytes()  - Integer.BYTES);
```